### DISCUSSION TITLE
Shouldn't Pyright fail subclasses that don't override `@abstractmethod`?

### DISCUSSION THREAD

**Discussion by Kache** [microsoft/pyright#9368] [Q&A] [answered]
According to https://docs.python.org/3.13/library/abc.html#abc.abstractmethod

>  A class that has a metaclass derived from ABCMeta cannot be instantiated unless all of its abstract methods and properties are overridden.

Code sample from #644 in [pyright playground](https://pyright-play.net/?code=JYWwDg9gTgLgBAQwEYGMBQGUBsEGddwCCAFMigHSEBCAwgJQBcacLcAAmecrjFAijBABTGAAsIAE2asJQgGZwA5iOK4hWOQBpEURQEYGcHlG0JdAJkNIIELI2mtWYPLkw58cKsUKMWAYjghAA8wIQEhCTgYCDg5BGAsKIBPUJRRMIBrBzhZBQA7CBgAfWUYVXU5XzAoYDyygHIAeQz6ugw0bBc4Gm97R1ylFTUNU10DI14qmrrieoBlGATE5tagA)

```python
import abc


class A(abc.ABC):
    @abc.abstractmethod
    def get(self, arg1: str, arg2: bool):
        pass

class B(A):  # expected to fail typecheck
    def not_get(self): print('Ok')


class C(A):
    def get(self, arg1: str): print('Still Ok')
```

`C` fails as expected, but `B` doesn't
Is this a bug?

![Image](https://github.com/user-attachments/assets/9c89f156-56a5-43da-894a-2acacc89fa1b)

**Comment by erictraut** [comment, marked as answer]
Pyright is working correctly here. The class definition of `B` is fine. You can create subclasses of `B` that provide the missing abstract method implementation.

If you attempt to instantiate `B`, pyright will report an error. Also, if you mark `B` as `@final`, then pyright will report an error.

**Comment by Kache** [comment]
Oh, I see, thanks

In my particular case, I had something like:

```py
class B(A):
    @classmethod
    def many(cls, n: int):
        for _ in range(n):
            yield cls()
            # yield B()
```

And `B()` does indeed fail, and I originally expected `cls()` to also, but that could be referring to a valid subclass
